kong/plugins/aws-lambda/iam-role-credentials.lua

```diff
@@ -0,0 +1,91 @@
+local http = require 'resty.http'
+local json = require 'cjson'
+local cache = require "kong.tools.database_cache"
+
+local CACHE_IAM_INSTANCE_CREDS_DURATION = 60 -- seconds to cache credentials from metadata service
+    
+local function get_iam_credentials_cache_key()
+  return "plugin.aws-lambda-iam-role.iam_role_temp_creds"
+end
+
+
+local function fetch_iam_credentials_from_metadata_service(metadata_service_host, metadata_service_port)
+    metadata_service_host = metadata_service_host or '169.254.169.254'
+    metadata_service_port = metadata_service_port or 80
+ 
+    local client = http.new()
+    local ok, err = client:connect(metadata_service_host, metadata_service_port) 
+    
+    if not ok then
+      ngx.log(ngx.ERR, "Could not connect to metadata service: " .. err)
+      return nil, err
+    end
+    
+    local role_name_request_res, err = client:request {
+      method = "GET",
+      path = "/latest/meta-data/iam/security-credentials/"
+    }
+     
+    if not role_name_request_res or role_name_request_res == "" then
+          ngx.log(ngx.ERR, "Could not fetch role name from metadata service: ".. err)
+        return nil, err
+    end 
+     
+    if role_name_request_res.status ~= 200 then
+      return nil, "[aws-lambda] Fetching role name from metadata service returned status code ".. role_name_request_res.status  .. 
```

Comment:
style: can we try to adjust line breaks and such so as to avoid the line being much longer than 80 chars